compiler/AST/ForallStmt.cpp

```diff
@@ -26,36 +26,106 @@
 
 /////////////////////////////////////////////////////////////////////////////
 //
+// ForallIntent represents one forall intent in a with clause
+//
+// This may be a transitional measure on the way to LoopIntentVar.
```

Comment:
I don't understand why we made ForallIntent instead of just immediately introducing LoopIntentVar or TaskIntent or something along those lines that is fairly general in name. However I'm OK with it "transitionally".